geomesa-spring/geomesa-spring-cloud-stream-binder-kafka-datastore/src/main/java/org/locationtech/geomesa/spring/binder/kafka/datastore/KafkaDatastoreBinderProvisioner.java

```diff
@@ -0,0 +1,52 @@
+/***********************************************************************
+ * Copyright (c) 2013-2024 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.spring.binder.kafka.datastore;
+
+import org.springframework.cloud.stream.binder.ConsumerProperties;
+import org.springframework.cloud.stream.binder.ProducerProperties;
+import org.springframework.cloud.stream.provisioning.ConsumerDestination;
+import org.springframework.cloud.stream.provisioning.ProducerDestination;
+import org.springframework.cloud.stream.provisioning.ProvisioningProvider;
+
+public class KafkaDatastoreBinderProvisioner implements ProvisioningProvider<ConsumerProperties, ProducerProperties> {
+
+    @Override
+    public ProducerDestination provisionProducerDestination(
+            final String name,
+            final ProducerProperties producerProperties) {
```

Comment:
i'm not entirely sure where these are coming from, but you should be able to pass producer/consumer configs down to the kds if you wanted